retrievalmarket/README.md

```diff
@@ -0,0 +1,124 @@
+# How to use the RetrievalMarket module
+
```

Comment:
I would provide a bit of a high level description of the basic architecture here:
- We provide a module with high level api's to execute retrievals
- We depend on the node to execute chain operations
- We abstract the things we need from the node into some high level interfaces.
- The node implementer must provide an implementation of these interfaces to construct the module

I would also at the end provide a brief "how to" for constructing a client & and a provider, and a how to call the client to query and then make retrieval deals.